composite/synth-index/src/adapter.ts

```diff
@@ -0,0 +1,54 @@
+import { Validator, Requester } from '@chainlink/external-adapter'
+import { AdapterResponse, Execute, AdapterRequest } from '@chainlink/types'
+import {
+  types,
+  makeExecute as tokenAllocationMakeExecute,
+} from '@chainlink/token-allocation-adapter'
+import snx from 'synthetix'
+import { makeConfig, Config } from './config'
+
+const customParams = {
+  base: ['base', 'asset', 'from'],
+  network: false,
+}
+
+type Synth = {
+  name: string
+  asset: string
+  index: any[]
+  inverted: Record<string, any>
+}
+
+const getAllocations = (synth: Synth): types.TokenAllocations => {
+  return synth.index.map((index) => ({
+    symbol: index.asset,
+    balance: index.units,
+    decimals: 0,
+  }))
+}
+
+export const execute = async (input: AdapterRequest, config: Config): Promise<AdapterResponse> => {
+  const validator = new Validator(input, customParams)
+  if (validator.error) throw validator.error
+
+  const jobID = validator.validated.jobID
+  const base = validator.validated.data.base.toLowerCase()
+  const network = validator.validated.data.network || config.defaultNetwork
+
+  const synths: Synth[] = snx.getSynths({ network: network.toLowerCase() })
+  const synth = synths
+    .filter(({ index, inverted }) => index && !inverted)
+    .find((d) => d.name.toLowerCase() === base)
+
+  if (!synth) {
+    return Requester.errored(jobID, new Error('Synth not found'))
+  }
```

Comment:
This should throw, not return an error response:


```suggestion
  if (!synth) throw new Error('Synth not found')
```